simulator/include/sim/simulator.hpp

```diff
@@ -4,21 +4,30 @@
 #include <sim/common.hpp>
 #include <sim/hart.hpp>
 #include <sim/memory.hpp>
+#include <sim/instr.hpp>
 
 namespace sim {
 
-struct Simulator final {
+class Simulator final {
+  public:
     enum class SimStatus {
         OK,
+        ERROR,
+        NOT_IMPLEMENTED_INSTR,
+        PHYS_MEMORY_ERROR,
+        PC_ALIGN_ERROR,
     };
 
   private:
-    memory::PhysMemory m_phys_memory;
+    memory::PhysMemory &m_phys_memory;
 
     hart::Hart m_hart{m_phys_memory};
 
   public:
-    template <class It> void binLoad(PhysAddr base_addr, It begin, It end);
+    Simulator(hart::Hart &hart, memory::PhysMemory &memory) : m_hart(hart), m_phys_memory{memory} {};
```

Comment:
Maybe it will be better to default initialize hart and memory? We can add getters for Hart& and PhysMemory& to Simulator API, thus user will be able to setup state before simulation